packages/web-components/src/stories/astro-uxds/core-concepts.stories.mdx

```diff
@@ -0,0 +1,144 @@
+for customization: CSS Custom Properties and CSS Shadow Parts.
+
+## Our Public API
+
+All of our Astro Web Components expose a public API, similar to native HTML elements. When using our web components,
+you should only interface with our public APIs or else your applications may unexpectedly break when upgrading. For example,
+avoid directly manipulating anything inside of the Shadow Root.
+
+Our public API includes the following: 
+
+* Properties/Attributes
+* Slots
+* Methods
+* Events
+* CSS Custom Properties (Styling)
+* CSS Shadow Parts (Styling)
+
+### Properties/Attributes 
+
+Properties and attributes define a component's appearance or behavior.
+
+### Slots
+
+Slots are placeholder sections within the Shadow DOM, allowing you full control over both the markup and styling of the content inside of it. 
+Each component will have all of its available slots documented under the API section. To make use of slots, use the `slot` attribute.
+
+For example: 
+
+```html
+<rux-container>
+    <header slot="header">My Header</header>
+    ... body content ...
+</rux-container>
+```
+
+```css
+header {
+    background: red;
+}
+```
+
+
+#### Default Slots
+
+Components may also have one unnamed default slot. These do not require the use of the `slot` attribute in your markup. 
+
+For example: 
+
+```html
+<rux-container>
+    <div id="this-is-default-slotted-content">
+        Hello World
+    </div>
+</rux-container>
+```
+
+```css
+#this-is-default-slotted-content {
+    background: red;
+}
+```
+
+#### When To Use
+
+Because slotted content is outside of the Shadow DOM, it can be styled just like any other element. You should slots 
+whenever you need full control over the content inside. For example, you may want to use the `label` slot on `rux-input`
+to append an icon to the label.
+
+On the other hand, if you only need a basic text label, you might want to just use the `label` attribute instead. 
+Some of our components offer both attributes and slots. In these scenarios, the attributes are less verbose and easier
+to write. 
+
+**With attribute:**
+```html
+<rux-input type="text" label="First Name"></rux-input>
+```
+
+**With slot:**
+```html
+<rux-input type="text">
+    <span slot="label">First name</span>
+</rux-input>
+```
+
+These two examples render the same output, but the first is significantly less work.
+
+### Methods 
+
+Methods are public functions of a component. They are commonly used to execute tasks or operations, for example, programatically setting a tree node to expand. All of our public methods are asynchronous and return a promise, which can be useful in scenarios where you need to perform additional actions after execution. 
+
+### Events 
+
+Like normal [Events](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events), Astro custom events are fired when a user interacts with an element. 
```

Comment:
Nit: call out react quirks here maybe? Link to react our react integration section